Assets/_Laz/Scripts/Managers/CheckPointManager.cs

```diff
@@ -1,28 +1,85 @@
-using System.Collections.Generic;
+using PerigonGames;
 using UnityEngine;
 
 namespace Laz
 {
     public class CheckPointManager : MonoBehaviour
     {
-        [SerializeField] private List<Transform> _checkPoints = null;
-        public Transform LastCheckPoint { get; private set; } = null;
+        [SerializeField] private Checkpoint[] _checkpoints;
+        private Checkpoint _activeCheckpoint = null;
+        private Vector3 _initialCheckPointPosition = Vector3.zero;
 
         public void Initialize()
         {
-            UpdateCheckPoint();
+            if (_checkpoints.IsNullOrEmpty())
+            {
+                return;
+            }
+            
+            for (int i = 0; i < _checkpoints.Length; i++)
+            {
+                Checkpoint checkpoint = _checkpoints[i];
+
+                if (checkpoint == null)
```

Comment:
This should throw an error. 
we don't want null checkpoints on this manager

Reply:
Throw Another error